cmd/dialer/main.go

```diff
@@ -0,0 +1,86 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"io"
+	"log/slog"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/getlantern/golog"
+	"github.com/getlantern/lantern-water/dialer"
+	waterDownloader "github.com/getlantern/lantern-water/downloader"
+	waterVC "github.com/getlantern/lantern-water/version_control"
+
+	_ "github.com/refraction-networking/water/transport/v1"
+)
+
+func main() {
+	ctx := context.Background()
+	log := slog.New(slog.NewTextHandler(os.Stdin, nil))
+
+	var listenerAddr, wasmAvailableAt, transportName string
+	flag.StringVar(&listenerAddr, "proxyURL", "localhost:8080", "URL of the proxy")
+	flag.StringVar(&wasmAvailableAt, "wasmAvailableAt", "https://github.com/getlantern/watm/releases/download/0.0.1/plain.v1.tinygo.wasm", "URL where the WASM is available")
```

Comment:
I would just use the mainline `github.com/refraction-networking/watm`.

Reply:
The mainline in this case doesn't provide a v1 WASM file, that's why I've created the `getlantern/watm` repository for holding some custom WASM bins (those bins are basically the examples they've provided)

Reply:
🤔  the utls could be added to [wateringhole](https://github.com/getlantern/wateringhole) and we should probably refer to the WASM file from there

Reply:
> it would be cool if wateringhole had a pipeline for generating those WASM files at releases, it would need to check each protocol implementation (if it's rust or go)

This sounds nice!  Probably not enough of a priority at the moment, but at some point I could see us setting up a CI job or cron job to handle this.